html/classhelper.js

```diff
@@ -178,6 +178,14 @@ class ClassHelper extends HTMLElement {
                 });
         };
 
+        const handlePopupReadyEvent = (event) => {
+            // we get a document Fragment in event.detail we replace it with the root
+            // replaceChild method consumes the documentFragment content, subsequent calls will be no-op.
+            if (e.detail.childElementCount === 1) {
```

Comment:
@rouilj I have extensively checked this behavior in all browsers and subsequent calls will not have the documentFragment with content inside as it's consumed here with replaceChild.

Thus eliminating our need to debounce and worry about document fragments need to be destroyed.

replaceChild consumes the fragment content leaving it empty, and the reference to it also gets cleaned up when either we remove the listener or popup window is closed